src/animation.js

```diff
@@ -130,13 +132,39 @@
         return 'pending';
       if (this.paused)
         return 'paused';
-      if (this.finished)
+      if (this._isFinished)
         return 'finished';
       return 'running';
     },
+    get finished() {
+      if (!this._finishedPromise) {
+        this._finishedPromise = new Promise(
+          function(resolve, reject) {
+            this.resolveFinished = resolve;
```

Comment:
This feels awkward but I don't know a better way to solve it :)

I'd maybe suggest a helper class since you're duplicating work here.

```js
if ('Promise' in window) {
  var PromiseWrapper = function() {
    this._promise = new Promise(function(resolve, reject) {
      this.resolve = resolve;
      this.reject = reject;
    }.bind(this));
  };
  PromiseWrapper.prototype = {
    get: function() { return this._promise; }
  };
} else {
  console.warn('Web Animations: Promise not available');
  var PromiseWrapper = function() {};
  PromiseWrapper.prototype = {
    resolve: function() {},
    reject: function() {},
    get: function() { return null; }
  };
}
```